### PR TITLE
Kustomize support for Argo Applications

### PR DESCRIPTION
**Pull request by spkane**
### What does this PR do?

Add support for installing `kustomize` based application via the blueprint's `argocd_application` map.

### Motivation

https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/621

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [x] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [x] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully completed before the merge?

### Additional Notes

- I was able to add this PR and  re-apply my existing `terraform` code with no changes.
- I was able to add the following block to my `argocd_application` object:

``` hcl
    "kustomize-test-${var.zone}" = {
      path     = "workloads/testing/litmus-chaos/overlays/${var.zone}"
      repo_url = local.addon_repos.internal_addons.url
      project  = "default"
      type     = "kustomize"
    },
```

which pointed to this file:

CleanShot 2022-06-16 at 16 00 17

and resulted in this application being added to ArgoCD.

CleanShot 2022-06-16 at 16 04 09

### CODE REVIEW

File: examples/gitops/argocd/main.tf [hunk 1]
```diff
@@ -91,6 +91,13 @@ module "eks_blueprints_kubernetes_addons" {
       repo_url           = "https://github.com/aws-samples/eks-blueprints-workloads.git"
       add_on_application = false
     }
+    /* other_workloads = {
+      path               = "kustomize-envs/overlays/dev"
+      # At the moment there are no kustomize examples in the aws-samples repo.
+      repo_url           = "https://github.com/aws-samples/eks-blueprints-workloads.git"
+      add_on_application = false
+      type               = "kustomize"
+    } */
```
Comment (askulkarni2) [inline examples/gitops/argocd/main.tf:100]: Please remove this block from here. The doc update should be enough. Maybe add a link for a [sample kustomize app](https://github.com/argoproj/argocd-example-apps/tree/master/kustomize-guestbook) in the docs for reference.

**Review by askulkarni2 — COMMENTED** [review]
Thanks for the PR. Looking good, one minor change requested.

**Comment by spkane** [comment]
Closes #621

File: examples/gitops/argocd/main.tf (hunk 1 above)
Comment (spkane) [reply examples/gitops/argocd/main.tf]: Done @askulkarni2 . Let me know if there is anything else that you want/need before it can be merged.

**Review by askulkarni2 — APPROVED** [review]
LGTM!

**Review by bryantbiggs — APPROVED** [review]
Thanks @spkane !

**Comment by spkane** [comment]
> Thanks @spkane !

No problem. Happy to help!

**Pull request merged**